### PR TITLE
General maintenance

### PR DESCRIPTION
**Pull request by selamanse**
This overall maintenance change does following:

- Rewrites the module to use a newer pve6,7 and 8 compliant api implementation https://github.com/luthermonson/go-proxmox
- Discards of unsused functionality for Deepshore
- Introduces some unit testing
- Introduces makefile for convenient overall build, test, deploy workflow
- Adds Deepshore to License
- Cleanup of code (lint, duplicates, wait and request efficiency)
- uses rancher compose instead of even older docker compose module
- Removes unsafe/deprecated crypto libs

# Tests

I have tested this module with following setup:
 - docker-machine version 0.16.2-gitlab.21, build 77bd8f58
 - mac m1 
 - PVE 7.4-17
 - given ansible playbook was also tested to create the base clone template

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,16 +1,35 @@
 module github.com/deepshore/docker-machine-driver-proxmoxve
 
-go 1.16
+go 1.19
```
Comment (tr-nextevolution) [inline go.mod]: Im github workflow wurde die Version auf 1.20 angehoben, warum hier nur 1.19?

File: driver.go [hunk 2]
```diff
@@ -1,44 +1,35 @@
...
 	*drivers.BaseDriver
-	driver *ProxmoxVE
-
-	// Top-level strategy for proisioning a new node
-	ProvisionStrategy string
+	client *proxmox.Client
 
 	// Basic Authentication for Proxmox VE
 	Host     string // Host to connect to
+	Port     string // Port to connect to (default 8006)
```
Comment (tr-nextevolution) [inline driver.go:32]: Warum kein int?

File: driver.go [hunk 3]
```diff
@@ -440,855 +401,463 @@ func (d *Driver) GetNetVlanTag() int {
...
 		return err
 	}
 
-	var stdoutBuf bytes.Buffer
-	session.Stdout = &stdoutBuf
-	session.Run("mkdir -p " + sshbasedir)
-	d.debugf(fmt.Sprintf("%s -> %s", hostname, stdoutBuf.String()))
-	session.Close()
+	_, task, err := clonevm.Clone(context.Background(), clone)
+	d.debugf("clone task for new vmid '%d' created. newId", newId)
```
Comment (tr-nextevolution) [inline driver.go]:
```suggestion
	d.debugf("clone task for new vmid '%d' created", newId)
```

**Review by tr-nextevolution — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (selamanse) [reply go.mod]: ah danke !

File: driver.go (hunk 2 above)
Comment (selamanse) [reply driver.go:32]: die uri wird in zeile 113 als string zusammengebaut.
da gibts vorab leider noch keine sinnvolle prüfung auf plausibilität.

**Pull request merged**